integration/integration_test.go

```
@@ -7891,6 +7893,325 @@ func getRemoteAddrString(sshClientString string) string {
+					OnLeave: string(types.OnSessionLeaveTerminate),
+				},
+			},
+		},
+	})
+	require.NoError(t, err)
+	_, err = aSrv.CreateRole(ctx, peerRole)
+	require.NoError(t, err)
+
+	peerUser, err := types.NewUser(peerUsername)
+	require.NoError(t, err)
+	peerUser.SetLogins([]string{username})
+	peerUser.SetRoles([]string{sshAccessRole.GetName(), peerRole.GetName()})
+	_, err = aSrv.CreateUser(ctx, peerUser)
+	require.NoError(t, err)
+
+	modUsername := username + "-moderator"
+	modRole, err := types.NewRole("moderator", types.RoleSpecV6{
+		Allow: types.RoleConditions{
+			JoinSessions: []*types.SessionJoinPolicy{{
+				Name:  "Session moderator",
+				Roles: []string{peerRole.GetName()},
+				Kinds: []string{string(types.SSHSessionKind)},
+				Modes: []string{string(types.SessionModeratorMode), string(types.SessionObserverMode)},
+			}},
+		},
+	})
+	require.NoError(t, err)
+	_, err = aSrv.CreateRole(ctx, modRole)
+	require.NoError(t, err)
+
+	modUser, err := types.NewUser(modUsername)
+	require.NoError(t, err)
+	modUser.SetLogins([]string{username})
+	modUser.SetRoles([]string{sshAccessRole.GetName(), modRole.GetName()})
+	_, err = aSrv.CreateUser(ctx, modUser)
+	require.NoError(t, err)
+
+	waitForNodesToRegister(t, instance, helpers.Site)
+
+	// Start a shell so a moderated session is created
+	peerTC, err := instance.NewClient(helpers.ClientConfig{
+		TeleportUser: peerUsername,
+		Login:        username,
+		Cluster:      helpers.Site,
+		Host:         Host,
+	})
+	require.NoError(t, err)
+
+	peerClusterCli, err := peerTC.ConnectToCluster(ctx)
+	require.NoError(t, err)
+	t.Cleanup(func() {
+		_ = peerClusterCli.Close()
+	})
+
+	nodeDetails := client.NodeDetails{
+		Addr:      instance.Config.SSH.Addr.Addr,
+		Namespace: peerTC.Namespace,
+		Cluster:   helpers.Site,
+	}
+	peerNodeCli, err := peerTC.ConnectToNode(
+		ctx,
+		peerClusterCli,
+		nodeDetails,
+		username,
+	)
+	require.NoError(t, err)
+	t.Cleanup(func() {
+		require.NoError(t, peerNodeCli.Close())
```

Why do you ignore errors when closing `peerClusterCli` but check them when closing `peerNodeCli`?

---

Ah that was a mistake, corrected